packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js

```diff
@@ -66,21 +72,22 @@ class DatePicker extends React.PureComponent {
 	}
 
 	isRangeInCurrentCalendar() {
-		const { startDate, endDate } = this.props;
+		const { startDate, endDate, selectedDate } = this.props;
 		if (!startDate && !endDate) {
 			return false;
 		}
 		return (
 			(startDate && this.isDateInCurrentCalendar(startDate)) ||
-			(endDate && this.isDateInCurrentCalendar(endDate))
+			(endDate && this.isDateInCurrentCalendar(endDate)) ||
+			this.isDateInCurrentCalendar(selectedDate)
```

Comment:
Let's say that you define a range, with the start date in january, and you select a date in march to set the end date. The dates in febuary won't be resulting as in range.

Reply:
ah yes 🤦‍♂